stactools_planet/stactools/planet/planet_item.py

```diff
@@ -125,44 +141,46 @@ def to_stac(self):
             bundle_type = planet_asset['annotations']['planet/bundle_type']
 
+            add_file_info = False
             # Planet data is delivered as COGs
-            if media_type == 'image/tiff' and asset_type not in [
-                    "udm", "udm2"
-            ]:
-                media_type = pystac.MediaType.COG
-                roles = ['visual']
-                thumbnail_path = f"{os.path.splitext(href)[0]}.thumbnail.png"
-                with rasterio.open(href) as dataset:
-                    height, width = dataset.shape
-                    geotransform = dataset.transform
-                    if width > height:
-                        width, height = 256, int(height / width * 256)
-                    else:
-                        width, height = int(width / height * 256), 256
-
-                    profile = dataset.profile
-                    profile.update(driver='PNG')
-                    profile.update(width=width)
-                    profile.update(height=height)
-
-                    if "analytic" in asset_type:
-                        data = dataset.read(indexes=[3, 2, 1],
-                                            out_shape=(3, height, width),
-                                            resampling=Resampling.cubic)
-                        profile.update(count=3)
-                    else:
-                        data = dataset.read(out_shape=(int(dataset.count),
-                                                       height, width),
-                                            resampling=Resampling.cubic)
-
-                    with rasterio.open(thumbnail_path, 'w', **profile) as dst:
-                        dst.write(data)
-
-                item.add_asset(
-                    'thumbnail',
-                    pystac.Asset(href=thumbnail_path,
-                                 media_type=pystac.MediaType.PNG,
-                                 roles=['thumbnail']))
+            if media_type == 'image/tiff':
+                add_file_info = True
+                if asset_type in ["udm", "udm2"]:
+                    roles = ['data']
+                else:
+                    media_type = pystac.MediaType.COG
+                    roles = ['visual']
+                    thumbnail_path = f"{os.path.splitext(href)[0]}.thumbnail.png"
+                    with rasterio.open(href) as dataset:
+                        height, width = dataset.shape
+                        geotransform = dataset.transform
+                        if width > height:
+                            width, height = 256, int(height / width * 256)
+                        else:
+                            width, height = int(width / height * 256), 256
+
+                        profile = dataset.profile
+                        profile.update(driver='PNG')
+                        profile.update(width=width)
+                        profile.update(height=height)
```

Comment:
FYI you can combine these into a single command if you want:
```py
>>> a = {}
>>> a.update(a=1, b=2)
>>> a
{'a': 1, 'b': 2}
```

Reply:
good catch. Will do it